build/utils-node.js

```diff
@@ -10,179 +10,540 @@
 var Q = require('q');
 var os = require('os');
 var vm = require('vm');
+var http = require('http');
+var url = require('url');
+var dive = require('diveSync');
+var nodeUUID = require('node-uuid');
+var jsdom = require('jsdom-nogyp').jsdom;
+var esprima = require('esprima');
+var procRunning = require('is-running');
 
-function joinPath() {
-  var src = path.join.apply(path, arguments);
-  console.log(src);
-  return src;
-}
-
-function getFile() {
-  var src = path.join.apply(path, arguments);
-  var fileStat;
-  try {
-    fileStat = fs.statSync(src);
-  } catch (e) {
-    // In order to have compability with xpcshell interface,
-    // we just simply let fileStat undefined and don't need
-    // to do anything here.
-  }
-  return {
-    exists: function() {
-      return !!fileStat;
-    },
-    isDirectory: function() {
-      return !!fileStat && fileStat.isDirectory();
-    },
-    isFile: function() {
-      return !!fileStat && fileStat.isFile();
-    },
-    path: src
-  };
-}
-
-function getFileContent(file, type) {
-  if (file.exists() && file.isFile()){
-    return fs.readFileSync(file.path, {encoding: type || 'utf-8'});
-  }
-}
+module.exports = {
 
-function Commander(cmd) {
+  Q: Q,
 
-  this.initPath = function(p) {};
+  scriptParser: esprima,
 
-  this.run = function(args, callback) {
-    var q = Q.defer();
-    var cmds = args.join(' ');
+  log: function() {
+    var args = Array.prototype.slice.call(arguments);
+    if (!args.length) {
+      console.log('\n');
+      return;
+    }
+    console.log('[' + args[0] + '] ' + args.slice(1).join(' '));
+  },
 
-    // In *nix and OSX version commands are run via sh -c YOUR_COMMAND,
-    // but in Windows commands are run via cmd /C YOUR_COMMAND,
-    // so, we just let execSync module to handle the difference.
-    if (cmd === 'sh') {
-      cmds = cmds.replace(/^-c/, '');
-    } else {
-      cmds = cmd + ' ' + cmds;
-    }
-    console.log(cmds);
-    // XXX: Most cmds should run synchronously, we should use either promise
-    //      pattern inside each script or find a sync module which doesn't
-    //      require recompile again since TPBL doesn't support that.
-    sh(cmds, function(err, stdout, stderr) {
-      if (err === null) {
-        callback && callback(0);
+  joinPath: function() {
+    var src = path.join.apply(path, arguments);
+    return src;
+  },
+
+  getOsType: function() {
+    var type = os.type();
+    return type === 'Windows_NT' ? 'WINNT' : type;
+  },
+
+  relativePath: function(from, to) {
+    return path.relative(from, to);
+  },
+
+  getFile: function() {
+    var src = path.join.apply(path, arguments);
+    var fileStat;
+    try {
+      fileStat = fs.statSync(src);
+    } catch (e) {
+      // In order to have compability with xpcshell interface,
+      // we just simply let fileStat undefined and don't need
+      // to do anything here.
+    }
+    return {
+      exists: function() {
+        return fs.existsSync(src);
```

Comment:
http://nodejs.org/api/fs.html#fs_fs_existssync_path 
fs.existsSync() will be deprecated, but I haven't found other method to replace it.
 Once refactor start, we should consider using promise to handle sync pb.